MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1357,114 @@ extension NavigationMapView {
     }
 }
 
+// MARK: - Building Extrusion Highlights
+
+extension NavigationMapView {
+    public func highlightBuildingExtrusion(for coordinate: CLLocationCoordinate2D) {
+        if let buildingId = buildingId(coordinate: coordinate) {
+            extrudeBuildings([(buildingId, buildingExtrusionColor)], extrudeAll: true)
+        } else {
+            unhighlightBuildingExtrusions()
+        }
+    }
+    
+    public func unhighlightBuildingExtrusions() {
+        extrudeBuildings([(0, buildingExtrusionColor)], extrudeAll: true)
+    }
+    
+    private func addStyleLayersIfNecessary() {
+        if let buildingsSource = style?.source(withIdentifier: "composite") {
+            
+            if let highlightedBuildingLayer = highlightedBuildingLayer {
+                style?.removeLayer(highlightedBuildingLayer)
+            }
+            
+            highlightedBuildingLayer = MGLFillExtrusionStyleLayer(identifier: StyleLayerIdentifier.buildingExtrusion, source: buildingsSource)
+            highlightedBuildingLayer?.sourceLayerIdentifier = "building"
+            highlightedBuildingLayer?.fillExtrusionColor = NSExpression(forConstantValue: UIColor.white)
+            highlightedBuildingLayer?.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+            highlightedBuildingLayer?.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+  
+            if let highlightedBuildingLayer = highlightedBuildingLayer {
+                style?.addLayer(highlightedBuildingLayer)
+            }
+        }
+    }
+
+    private func buildingId(coordinate: CLLocationCoordinate2D) -> Int? {
+        let screenCoordinate = convert(coordinate, toPointTo: self)
+        let features = visibleFeatures(at: screenCoordinate)
+        
+        if let feature = features.first, let buildingId = feature.identifier as? Int {
```

Comment:
By default, `MGLMapView.visibleFeatures(at:)` returns visible features in all layers at the screen coordinate, sorted from top to bottom in z order. However, this code considers only the first returned feature, regardless of whether it’s a building. This could cause the method to fail if, for instance, a POI label or destination annotation happens to overlap the query coordinate.

There are a couple ways to make this code more robust:

* Filter `MGLStyle.layers` to layers whose `sourceLayerIdentifier` is [`building`](https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/#building), then pass those layers into `visibleFeatures(at:styleLayerIdentifiers:)`. But this assumes the style shows buildings to begin with, which isn’t the case for Satellite Streets or some other likely styles.
* Add a faintly visible building layer to the style, not really visible to the naked eye but opaque enough for feature querying to find it. For example, this is how the current road name feature makes road name querying more robust using white lines: https://github.com/mapbox/mapbox-navigation-ios/blob/3f361bb8735bf2671dc302484e75feb964976a89/MapboxNavigation/RouteMapViewController.swift#L658-L665
* Filter the returned features for feature attributes that one would expect on a building, such as `extrude`.

Reply:
Turns out that it's possible to add only one `MGLFillExtrusionStyleLayer` with `sourceLayerIdentifier` set to `building`. In case if extrusion layer has different ID buildings will not be extruded. This means that only 3rd approach is fine in this case. I've added filtering which takes into account `extrude` and `type` attributes.